### PR TITLE
Extract framework into gem

### PR DESCRIPTION
**Pull request by ljones140**
Framework has been extracted into Gem. All tests are passing. 
Had to change `Time.zone.now` to `Time.now`. Advice welcome how to require the part of active support to use zone.

Coveralls added.
Rakefile set up to run all tasks.

Please sign off
@chmeldax @erishine

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,80 @@
+PATH
```
Comment (chmeldax) [inline Gemfile.lock]: Gemfile.lock shouldn't be part of the repo. It's not an app that is deployed somewhere.

File: kitcat.gemspec [hunk 2]
```diff
@@ -0,0 +1,21 @@
+Gem::Specification.new do |spec|
+  spec.name        = 'kitcat'
+  spec.version     = '0.0.0'
+  spec.summary     = 'a framework to support data processing'
+  spec.description = 'initially created for data migrations. Provides logging, progess bar and graceful handling'
+  spec.authors     = ['Simply Business']
+  spec.email       = ['tech@simplybusiness.co.uk']
+  spec.homepage    = 'https://github.com/simplybusiness/kitcat'
+  spec.license     = 'MIT'
+  spec.files       = Dir['lib/   *.rb']
```
Comment (chmeldax) [inline kitcat.gemspec]: Think about including other files (like readmes, tests etc.)

File: Gemfile.lock (hunk 1 above)
Comment (ljones140) [inline Gemfile.lock]: Gemfile.lock has been removed and git ignored

File: kitcat.gemspec (hunk 2 above)
Comment (ljones140) [inline kitcat.gemspec]: I've made changes to the gemspec in a following commit.

**Comment by joshuafleck** [comment]
Would be good to use rubocop and make it part of the semaphore run. More guidelines can be found here: https://github.com/simplybusiness/how-we-roll/blob/master/development/NEW_PROJECT_GUIDELINES.markdown#ruby-projects

**Comment by chmeldax** [comment]
@joshuafleck Yep, that stuff will come in a different PR, I am looking into it.

But, I guess I prefer much more TravisCI to Semaphore (not sure if this poses a problem for SB).

File: lib/kitcat.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+require 'require_all'
```
Comment (chmeldax) [inline lib/kitcat.rb:1]: Do we need this?

Comment (ljones140) [inline lib/kitcat.rb:1]: It makes the requiring of files easier. SB uses it in one of their popular gems

**Comment by chmeldax** [comment]
singed-off-by @chmeldax

File: lib/kitcat.rb (hunk 3 above)
Comment (ljones140) [inline lib/kitcat.rb:1]: I removed the dependancy before the merge

**Pull request merged**